metadata-jobs/mae-consumer/src/main/java/com/linkedin/metadata/kafka/MetadataChangeLogProcessor.java

```diff
@@ -41,61 +43,159 @@
 @EnableKafka
 public class MetadataChangeLogProcessor {
 
-  private final List<MetadataChangeLogHook> hooks;
   private final Histogram kafkaLagStats = MetricUtils.get().histogram(MetricRegistry.name(this.getClass(), "kafkaLag"));
 
+  private static final String GROUP_1 = "GROUP_1";
+  private static final String GROUP_2 = "GROUP_2";
+  private static final String GROUP_3 = "GROUP_3";
+  private static final String GROUP_4 = "GROUP_4";
+
+  private static final String HOOK_NAME_1 = "updateIndicesHook";
+  private static final String HOOK_NAME_2 = "ingestionSchedulerHook";
+  private static final String HOOK_NAME_3 = "entityChangeEventHook";
+  private static final String HOOK_NAME_4 = "siblingAssociationHook";
+
+  @Value("${kafka.mae.consumer.group.1:}")
```

Comment:
Changing the consumer groups has significant impact on existing production servers and could result in an entire reprocessing of the current Kafka log. Have to think about how we can do this without impacting existing deployments.

Reply:
@RyanHolstien Hm... you are right. This could be a problem. What about keeping the existing group and adding new groups?

Reply:
@RyanHolstien Could you have a look? I have reverted the existing group in order to avoid reprocessing - the code is refactored but it will do the same as earlier. 
New groups are added to support the new features. Note that the default group kafka listener does not throw an exception (it was like that earlier), but new new kafka listeners can throw an exception as we want the message to be redelivered again.
Also, all the default properties values are set to achieve the same behavior as we had before this PR. If the user wants to use new listener retry mechanism, he needs to set those properties 
```
kafka.mae.consumer.group.1
kafka.mae.consumer.group.2
kafka.mae.consumer.group.3
kafka.mae.consumer.group.4
```
and update this property
`kafka.mae.consumer.group.default`

Initial properties are:
```
  mae:
    consumer:
      group:
        default: ${KAFKA_MAE_CONSUMER_GROUP_DEFAULT:updateIndicesHook,ingestionSchedulerHook,entityChangeEventHook,siblingAssociationHook}
        1: ${KAFKA_MAE_CONSUMER_GROUP_1:}
        2: ${KAFKA_MAE_CONSUMER_GROUP_2:}
        3: ${KAFKA_MAE_CONSUMER_GROUP_3:}
        4: ${KAFKA_MAE_CONSUMER_GROUP_4:}
```